src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1553,6 +1568,10 @@ def extract_arg_descriptors(self):
                     assert arg_info.default in [gt_ir.Empty, None]
                     data_type = gt_ir.DataType.from_dtype(np.dtype(arg_annotation.dtype))
                     axes = [ax.name for ax in arg_annotation.axes]
+                    seq_name = gt_ir.Domain.LatLonGrid().sequential_axis.name
+                    if len(axes) == 1 and seq_name not in axes:
+                        # If there is only a single axis, it cannot be a parallel axis.
```

Comment:
Why is that? It should be fine as long as it's read only, no?

Reply:
The way I see it, the rule should be "if it has any parallel axis, it should have all parallel axes" since it will cause a race condition in other cases. This test doesn't quite reflect that statement yet. I'll make a change to it.

Reply:
Perhaps that should be amended to "cannot write to any field unless all parallel axes are present".